compiler/src/dotty/tools/dotc/transform/CheckUnused.scala

```diff
@@ -643,9 +643,21 @@ object CheckUnused:
         sel.isWildcard ||
         imp.expr.tpe.member(sel.name.toTermName).alternatives.exists(_.symbol.isOneOf(GivenOrImplicit)) ||
         imp.expr.tpe.member(sel.name.toTypeName).alternatives.exists(_.symbol.isOneOf(GivenOrImplicit))
-      )
-
+      ) 
+    
+    /**
+     * Ignore CanEqual imports
+     */
+    private def isImportIgnored(imp: tpd.Import, sel: ImportSelector)(using Context): Boolean =
+      (sel.isWildcard && imp.expr.tpe.allMembers.exists(p => p.symbol.typeRef.baseClasses.exists(_.derivesFrom(defn.CanEqualClass)))) ||
```

Comment:
I think this is problematic because perhaps it only imports CanEqual's that are not even used. Perhaps it makes more sense to record the can equal usages in the typer?

Reply:
Also this should check if its also a "given" import, rather than "*"

Reply:
Unfortunately, after a lengthy discussion with Martin and the team and making some PoCs, we decided that extracting that information from Typer is unfeasible without making the Typer and Implicit resolution slower / more complex. So we want to emulate it or fix false positives. But I will address the second commend 👍

Reply:
Yes, and this logic could get even more complex I suppose given that it is a type param, and the resolution is pre typer vs the emulation would be post typer. Not a good idea

Reply:
Im not sure how it works, but might it produce more accurate warnings if you only ignore warnings generated by the import selector when the specific unused symbol derives from CanEqual?

Reply:
It's pretty much what we are doing here